### ISSUE TITLE
Warning log level should capture full throwable stack trace

### ISSUE THREAD

**Issue by AlmasB** [AlmasB/FXGL#947]
`log.warning(message, e)`

**Comment by AlmasB** [comment]
@adambocco might be of interest?

**Comment by adambocco** [comment]
@AlmasB 

Do you just want this change made to the `warning(message, error)` and not `warning(message)`. If so something similar to what happens at the fatal level would be fine wouldn't it?

```kotlin
    fun warning(message: String, error: Throwable) {
        val trace = error.stackTraceToString()
        warning("$message\n$trace")
    }
```

**Comment by AlmasB** [comment]
> Do you just want this change made to the warning(message, error) and not warning(message)

Yes, since `warning(message)` is self-contained, whereas with `error` we know the cause, which should be printed in full.

As for the above solution, yes that's what I had in mind. I opted in for the single line print (as it is now), but that, sometimes, doesn't provide the necessary info behind the error. The full stacktrace might be an overkill (I'm particularly worried about `warning()` calls during the main loop, which could spam the console), but I don't think we have any other option. Besides, it's not like warning is the "normal case" and should only occur when there is something off, which developers would need to fix, so I think we can go ahead with above solution.

If you have any further thoughts on this, feel free to share.

**Comment by AlmasB** [comment]
Closed by #990
